### ISSUE TITLE
some question about loss

### ISSUE THREAD

**Issue by 905390215** [open-mmlab/mmsegmentation#1604]
lr: 4.516e-04, eta: 0:46:08, time: 1.019, data_time: 0.006, memory: 8574, decode.loss_ce: 0.0604, decode.acc_seg: 97.2657, aux.loss_ce: 0.0383, aux.acc_seg: 95.7123, loss: 0.0987

I am experimenting with the algorithm in mmseg,
 is the last loss of loss change (lr: 4.516e-04, eta: 0:46:08, time: 1.019, data_time: 0.006, memory: 8574, decode.loss_ce: 0.0604, decode.acc_seg: 97.2657, aux.loss_ce: 0.0383, aux.acc_seg: 95.7123, loss: 0.0987) in the experiment the training set loss value or the test set loss value? 
If it is the training set loss value, how can I get the test set loss value?

**Comment by MengzhangLI** [comment]
Currently MMSegmentation V0.24.1 has supported validation loss in log: https://github.com/open-mmlab/mmsegmentation/pull/1494.

Test set is not evaluated because of it may have no annotations. However, you could change test set (it has annotations) in config file, be like: https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/datasets/ade20k.py#L43-L54

**Comment by 905390215** [comment]
Sorry, my foundation is poor, I don't fully understand what you mean, I use the voc2012 data set, can you tell me how to modify it?

data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir='SegmentationClass',
        split='ImageSets/Segmentation/train.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir='SegmentationClass',
        split='ImageSets/Segmentation/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(2048, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),

**Comment by 905390215** [comment]
Sorry, my foundation is poor, I don't fully understand what you mean, I use the voc2012 data set, can you tell me how to modify it?

data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir='SegmentationClass',
        split='ImageSets/Segmentation/train.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir='SegmentationClass',
        split='ImageSets/Segmentation/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(2048, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),

**Comment by MengzhangLI** [comment]
The Validation loss is recorded in log if mmseg version >0.24.1. 

As for other problems read the tutorials first.

https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/tutorials/config.md

https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/tutorials/customize_datasets.md
